dev/ci.js

```diff
@@ -0,0 +1,82 @@
+import { createServer } from 'https'
+import { readFileSync } from 'fs'
+import { spawn, spawnSync } from 'child_process'
+import { once } from 'events'
+
+import { sendResponse } from './mocks.js'
+
+/*
+  every hours:
+    - git fetch
+    - cleanup stale database files (older than a week ?)
+
+*/
+
+const read = async (stream) => {
+  stream.setEncoding('utf8')
+  const chunks = []
+  for await (const chunk of stream) chunks.push(chunk)
+  return chunks.join('')
+}
+
+const handleRequest = async (req, res, again) => {
+  const hash = req.url.split('/', 2)[1]
+  const version = req.headers.referer?.match(
+    /https:\/\/([a-z0-9]{8})\.platform-au0l\.pages\.dev/,
+  )?.[1]
+  if (!version || !hash || hash.length !== 40) {
+    res.statusCode = 403
+    return res.end()
+  }
+  res.setHeader('access-control-allow-origin', '*')
+  res.setHeader(
+    'access-control-allow-headers',
+    'origin, x-requested-with, content-type, accept',
+  )
```

Comment:
classic CORS headers, needed since the domain of the API is different in that case.